src/mo_frontend/typing.ml

```diff
@@ -1800,10 +1800,13 @@ and check_pat' env t pat : Scope.val_env =
         display_typ_expand t
     in check_pat env t1 pat1
   | TagP (id, pat1) ->
-    let t1 =
-      match T.lookup_val_field_opt id.it (T.as_variant_sub id.it t) with
-      | Some t1 -> t1
-      | None -> T.Non
+    let t1 = try
+        match T.lookup_val_field_opt id.it (T.as_variant_sub id.it t) with
```

Comment:
```suggestion
     match T.lookup_val_field_opt id.it (T.as_variant_sub id.it t) with
```

Reply:
indenting by 1 seemed strange, how about this?